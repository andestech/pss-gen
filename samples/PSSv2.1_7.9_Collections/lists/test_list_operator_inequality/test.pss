/*!
@weakgroup      test
@{
@file
@author         Luther Lee
@data           2023/10/25
@brief          Test list inequality operator `!=`.
@version
Portable Test and Stimulus Standard Version 2.1 <br>
> Section 7.9 : Collections <br>
> Section 7.9.3 : Lists <br>


---------------------------------------------------------------------------------------------------

### Test Command
@code{.unparsed}
make gentarget
@endcode
*/
component pss_top {

    enum eSTR2NUM {
        ZERO, ONE, TWO
    };

    struct sSTR2NUM {
        string stringVal = "NULL";
        int intVal = 0;
    };

    action root_a {
        list<bit [4] > nibbleList = {4'b0010, 4'b0011};
        list<int     > intList    = {2, 3};
        list<bool    > boolList   ;
        list<eSTR2NUM> enumList   ;
        list<string  > stringList = {"A", "B"};
        list<float32 > float32List;
        list<float64 > float64List;
        // list<chandle > chandleList;
        list<sSTR2NUM> structList ;

        bit nibbleList_equ  = 0;
        bit intList_equ     = 0;
        bit boolList_equ    = 0;
        bit enumList_equ    = 0;
        bit stringList_equ  = 0;
        bit float32List_equ = 0;
        bit float64List_equ = 0;
        bit structList_equ  = 0;
        exec post_solve {
            if (nibbleList  != nibbleList ) nibbleList_equ  = 1;
            if (intList     != intList    ) intList_equ     = 1;
            if (boolList    != boolList   ) boolList_equ    = 1;
            if (enumList    != enumList   ) enumList_equ    = 1;
            if (stringList  != stringList ) stringList_equ  = 1;
            if (float32List != float32List) float32List_equ = 1;
            if (float64List != float64List) float64List_equ = 1;
            if (structList  != structList ) structList_equ  = 1;
        }

        exec body ASM =
"""
// test_list_operator_inequality
nibbleList  = {{nibbleList_equ}}
intList     = {{intList_equ}}
boolList    = {{boolList_equ}}
enumList    = {{enumList_equ}}
stringList  = {{stringList_equ}}
float32List = {{float32List_equ}}
float64List = {{float64List_equ}}
structList  = {{structList_equ}}
""";
    }
}

/**@}*/
