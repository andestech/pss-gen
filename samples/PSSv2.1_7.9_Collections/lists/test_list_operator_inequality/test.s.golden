
// test_list_operator_inequality
nibbleList  = 0
intList     = 0
boolList    = 0
enumList    = 0
stringList  = 0
float32List = 0
float64List = 0
structList  = 0

